packages/brain/src/modules/apiServers/launchpad/routes/feeRecipients.ts

```diff
@@ -0,0 +1,138 @@
+import express from "express";
+import { getValidators } from "../../../../calls/getValidators.js";
+import { CustomValidatorGetResponse, isValidBlsPubkey } from "@stakingbrain/common";
+import { BrainPubkeysFeeRecipients } from "../types.js";
+import { validateUpdateFeeRecipientRequestBody } from "../validation/requestValidation.js";
+import { updateValidators } from "../../../../calls/updateValidators.js";
+
+const feeRecipientsRouter = express.Router();
+
+const feeRecipientsEndpoint = "/eth/v1/feeRecipients";
+
+/**
+ * Retrieves fee recipient information for specified validators via GET request.
+ * Clients can filter validators by providing a list of BLS public keys (pubkeys) as
+ * a query parameter. If no pubkeys are provided, information for all validators is returned.
+ * Only valid BLS public keys are considered for filtering.
+ *
+ * Example usage:
+ * GET /api/feeRecipients?pubkeys=0x12345,0x67890
+ * 
+ * This request returns fee recipient details for validators with public keys 0x12345 and 0x67890.
+ *
+ * Example response:
+ * {
+ *   "validators": [
+ *     {
+ *       "pubkey": "0x12345",
+ *       "feeRecipient": "0xabcdef"
+ *     },
+ *     {
+ *       "pubkey": "0x67890",
+ *       "feeRecipient": "0x123456"
+ *     }
+ *   ]
+ * }
+ */
+feeRecipientsRouter.get("/eth/v1/feeRecipients", async (req, res) => {
+    try {
+        const validators: CustomValidatorGetResponse[] = await getValidators();
+        const pubkeysParam = req.query.pubkeys as string | undefined;
+
+        if (!pubkeysParam) {
+            // If no pubkeys are provided, return information for all validators
+            const response: BrainPubkeysFeeRecipients = {
+                validators: validators.map(validator => ({
+                    pubkey: validator.pubkey,
+                    feeRecipient: validator.feeRecipient
+                }))
+            };
+
+            return res.status(200).json(response);
+        }
+
+        const pubkeys = pubkeysParam.split(',').map(pubkey => pubkey.toLowerCase());
+        const invalidPubkeys = pubkeys.filter(pubkey => !isValidBlsPubkey(pubkey));
+
+        if (invalidPubkeys.length > 0) {
+            return res.status(400).send({ message: `Invalid pubkey format: ${invalidPubkeys.join(", ")}. Pubkeys should follow BLS format (beginning with 0x)` });
+        }
+
+        const filteredValidators = validators.filter(validator => pubkeys.includes(validator.pubkey.toLowerCase()));
```

Comment:
would it be better to return a 404 not found if filteredvalidators is empty?

Reply:
You are right